src/containers/Auth/Signin.tsx

```diff
@@ -0,0 +1,132 @@
+import {
+  ChangeEvent,
+  FocusEvent,
+  SyntheticEvent,
+  useEffect,
+  useState,
+} from 'react';
+import { useRouter } from 'next/router';
+import useRequest from '@/hooks/useRequest';
+import Button from '@/components/Button';
+import Header from './components/Header';
+import InputContainer from './components/InputContainer';
+import Social from './components/Social';
+import { ERROR_MESSAGES, validateEmail, validatePassword } from './validation';
+
+interface Signin {
+  data: {
+    accessToken: string;
+    refreshToken: string;
+  };
+}
+
+function Signin() {
+  const router = useRouter();
+  const [emailErrorMessage, setEmailErrorMessage] = useState('');
+  const [passwordErrorMessage, setPasswordErrorMessage] = useState('');
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+
+  const onEmailChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const nextValue = e.target.value;
+    setEmail(nextValue);
+  };
+
+  const onPasswordChange = (e: ChangeEvent<HTMLInputElement>) => {
+    const nextValue = e.target.value;
+    setPassword(nextValue);
+  };
```

Comment:
해당 함수들이 비슷한 형식으로 반복된다는 느낌을 받았지만, button의 onclick 함수로 내려주어야하기에 통일하지 못했습니다. 혹시 더 간결하게 나타낼 수 있는 방법이 있을 지 궁금합니다.

Reply:
`해당 함수들이 비슷한 형식으로 반복된다는 느낌을 받았지만` <- 아주 좋은 느낌입니다!
`button의 onclick 함수로 내려주어야하기에 통일하지 못했습니다` <- 이건 어떤 의미인지 잘 모르겠네요. 🤔 이벤트 객체 (`e`)를 인자로 받아야 해서, setState 함수를 인자로 넣어주기 어렵다는 의미일까요?

제가 생각하는 해결책은 다음과 같습니다.
1. 굳이 별도의 공통 함수를 만들지 않고, onChange에 이벤트 객체를 인자로 받는 `익명 함수`를 넣어주는 것입니다.  onChange와 setState함수를 이용해 값을 업데이트를 해주는 작업은 리액트에서 흔한 패턴이라 익명함수를 사용해도 가독성에 문제가 없을 것으로 보입니다. 실제로 코드가 간결하기도 하구요.

```suggestion
    <InputContainer
      // (...)
      onChange={(e) => setEmail(e.target.value)}
    >
      이메일
    </InputContainer>
    <InputContainer
      // (...)
      onChange={(e) => setPassword(e.target.value)}
    >
      비밀번호
    </InputContainer>
    >
```

2. 그래도 익명함수 대신 별도의 함수를 만들기 원하신다면 먼저 여러개의 state를 하나로 통합하는 것이 좋아보이네요. 결국 하나의 함수로 통합할 때 각 함수에서 사용하는 setState 함수가 다르기 때문에 곤란한 부분이 발생하는 것이여세요. 공통 함수에서는 하나의 state를 변경하되 이벤트 객체에서 각 input을 구분할 수 있는 정보를 받아오도록 하면 좋을 것 같습니다. (예를 들면 input에 name 어트리뷰트를 설정한다던가 해서요.) 🤗